src/components/search/data/hooks.js

```diff
@@ -107,19 +108,48 @@ export const useNbHitsFromSearchResults = (searchResults) => {
   return nbHits;
 };
 
-export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan }) => {
+export const getCatalogString = (offerCatalogs, initialString = '') => {
+  const offerCatalogsCopy = [...offerCatalogs];
+  const lastCatalog = offerCatalogsCopy.pop();
+  return `${offerCatalogsCopy.reduce(
+    (result, catalog) => `${result}enterprise_catalog_uuids:${catalog} OR `, initialString,
+  )}enterprise_catalog_uuids:${lastCatalog}`;
+};
+
+export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan, offerCatalogs = [] }) => {
+  // default to showing all catalogs
+  const { showAllCatalogs, setShowAllCatalogs } = useContext(SearchContext);
+
+  useMemo(() => {
+    // if there are no subscriptions or offers, we default to showing all catalogs
+    if (!subscriptionPlan && offerCatalogs.length < 1) {
+      setShowAllCatalogs(true);
+    }
+  }, [subscriptionPlan, offerCatalogs.length]);
+
   const filters = useMemo(
     () => {
+      if (showAllCatalogs) {
+        if (subscriptionPlan) {
+          // show subscription catalog and all other enterprise catalogs
+          return `enterprise_catalog_uuids:${subscriptionPlan.enterpriseCatalogUuid} OR enterprise_customer_uuids:${enterpriseConfig.uuid}`;
```

Comment:
Do we need to filter on both `enterprise_catalog_uuids` and `enterprise_customer_uuids` here? The subscription catalog would be part of the catalogs that come back with the `enterprise_customer_uuids` filter.

Reply:
Unfortunately we do. I also thought it would come back with the `enterprise_customer_uuids`, but it didn't, so I ended up with this. I'll test it one more time, though, it's possible I was doing something else wrong at the time.

Reply:
A course should always be tagged with both the associated `enterprise_customer_uuids` and `enterprise_catalog_uuids`. I'm definitely surprised if the subs catalog doesn't come back with the `enterprise_customer_uuids` filter alone.

I wonder if the Algolia index you're working with locally didn't add the appropriate enterprise uuid to the list of `enterprise_customer_uuids` associated with the course(s) for some reason?

Reply:
It turns out I was doing something different, and this works and simplifies the logic, so yay!